test/curve_fit.jl

```diff
@@ -1,6 +1,6 @@
 let
     # fitting noisy data to an exponential model
-    model(x, p) = p[1]*exp(-x.*p[2])
+    model(x, p) = @compat p[1] .* exp.((-).(x) .* p[2])
```

Comment:
I'd always make the opposite sacrifice for a 2x reduction of allocation especially for non-public API but I guess this is only the test so I don't care. I changed it to the slower version and left TODO's for how to update it in the future.